org.adempiere.base/src/org/compiere/util/DB.java

```diff
@@ -2410,22 +2410,54 @@ public static void createT_Selection(int AD_PInstance_ID, Collection<Integer> se
 	 * @param AD_PInstance_ID
 	 * @param saveKeys
 	 * @param trxName
+	 * @deprecated Preserved for Backward Compatibility, use createT_SelectionNewNP instead
 	 */
-	public static void createT_SelectionNew (int AD_PInstance_ID, Collection<KeyNamePair> saveKeys, String trxName)
+	public static void createT_SelectionNew (int AD_PInstance_ID, Collection<KeyNamePair> saveKeys, String trxName) {
+		Collection<NamePair> saveKeysNP = new ArrayList<NamePair>();
+		for (NamePair saveKey : saveKeys)
+			saveKeysNP.add(saveKey);
+		createT_SelectionNewNP(AD_PInstance_ID, saveKeysNP, trxName);
+	}
+
+	/**
+	 * Create persistent selection in T_Selection table
+	 * saveKeys is map with key is rowID, value is list value of all viewID
+	 * viewIDIndex is index of viewID need save.
+	 * @param AD_PInstance_ID
+	 * @param saveKeys can receive a collection of KeyNamePair (IDs) or ValueNamePair (UUIDs)
+	 * @param trxName
+	 */
+	public static void createT_SelectionNewNP (int AD_PInstance_ID, Collection<NamePair> saveKeys, String trxName)

```

Comment:
I'm not sure it is a good idea of sharing the method here for ValueNamePair and KeyNamePair. This make it a case of depends on the Developer doing the right thing - the method signature allows a collection that includes both ValueNamePair and KeyNamePair although that wouldn't works correctly.

Reply:
Thanks @hengsin - I changed the method to be more safe when receiving objects ValueNamePair and KeyNamePair mixed.  Also, it raises an error in case there is some NamePair not within those two classes.

I did tests to the modified method inserting 1000, 2000, 2500 records, and it's working fine.

The QueryTest unit test is also working fine - I reverted the deprecation of createT_SelectionNew, as it is still a valid method to work with KeyNamePair collections.